earthaccess/search.py

```diff
@@ -468,8 +468,7 @@ def get(self, limit: int = 2000) -> List[DataGranule]:
             RuntimeError: The CMR query failed.
         """
         response = get_results(self.session, self, limit)
-
-        cloud = self._is_cloud_hosted(response[0])
+        cloud = response and self._is_cloud_hosted(response[0])
```

Comment:
MyPy is complaining here so I think this needs to be:
```suggestion
        cloud = len(response) > 0 and self._is_cloud_hosted(response[0])
```
unless we can tell MyPy what we are doing.

Reply:
Oh, itneresting, MyPy is right here:
```python
>>> response = []
>>> cloud = results and isinstance(response[0], dict)
>>> print(cloud)
[]
>>> response = [{'foo': 'bar'}]
>>> cloud = results and isinstance(response[0], dict)
>>> print(cloud)
True
```